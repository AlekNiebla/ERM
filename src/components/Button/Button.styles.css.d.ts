export const main: string;
export const label: string;
export const grey: string;
export const primary: string;
export const negative: string;
export const disabled: string;
