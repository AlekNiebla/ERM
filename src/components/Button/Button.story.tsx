import * as React from "react";
import { storiesOf } from "@storybook/react";
import Button from "./Button";

storiesOf("Components / Button", module)
  .add("default", () => <Button label="default" />)
  .add("primary", () => <Button label="primary" type="primary" />)
  .add("negative", () => <Button label="negative" type="negative" />);
