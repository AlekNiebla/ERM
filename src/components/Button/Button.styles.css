.main {
  padding: 10px 20px;
  border-radius: 4px;
  border-style: none;
  color: var(--color-primary);
}
.label {
  composes: Typo-Base from global;
}
.grey {
  background-color: var(--color-gray-pale);
}
.primary {
  background-color: var(--color-primary);
  color: var(--color-white);
}
.primary:hover,
.grey:hover {
  background-color: var(--color-secondary);
  color: var(--color-primary);
}
.primary:active,
.grey:active {
  background-color: var(--color-secondary-darker);
  color: var(--color-primary);
}
.negative {
  background-color: var(--color-negative-darker);
  color: var(--color-white);
}
.negative:hover {
  background-color: var(--color-negative);
}
.negative:active {
  background-color: var(--color-negative-lighter);
}
.disabled {
}
