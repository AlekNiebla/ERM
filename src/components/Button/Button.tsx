import * as React from "react";

interface IButton {
  label: string;
}

const Button = (props: IButton) => <button>{props.label}</button>;

export default Button;
