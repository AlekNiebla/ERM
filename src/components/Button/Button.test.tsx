import * as React from "react";
import { shallow } from "enzyme";
// components
import Button from "./Button";

describe("Button component...", () => {
  it("...renders without errors", () => {
    const label = "test label";

    const wrapper = shallow(<Button label={label} />);

    expect(wrapper.find(Button)).toBeTruthy();
    expect(wrapper.contains(label)).toBeTruthy();
    // snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
