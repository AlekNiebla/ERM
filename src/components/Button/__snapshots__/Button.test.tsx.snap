// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component... ...renders without errors 1`] = `ShallowWrapper {}`;
